application/aam-backend-service/src/main/kotlin/com/aamdigital/aambackendservice/couchdb/core/CouchDbInitializer.kt

```diff
@@ -0,0 +1,69 @@
+package com.aamdigital.aambackendservice.couchdb.core
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.InitializingBean
+
+
+/**
+ * Representing a couchdb database
+ */
+open class DatabaseRequest(
+    val name: String
+)
+
+/**
+ * Service to initialize the couchdb instance.
+ * Will create all databases needed. Can be extended by feature modules.
+ * If a feature is getting enabled later on, the database is created then.
+ *
+ * No databases are deleted by disabling features.
+ *
+ */
+class CouchDbInitializer(
+    private val couchDbClient: CouchDbClient,
+    private val databaseRequests: List<DatabaseRequest>
+) : InitializingBean {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    companion object {
+        private val DEFAULT_DATABASES = listOf(
+            DatabaseRequest("app"),
+            DatabaseRequest("app-attachments"),
+        )
+    }
+
+    private fun initCouchDb() {
+        DEFAULT_DATABASES.forEach {
+            createDatabase(it)
+        }
+
+        databaseRequests.forEach {
+            createDatabase(it)
+        }
+    }
+
+    private fun createDatabase(databaseRequest: DatabaseRequest) {
```

Comment:
Possibly we want to reuse this (programmatically triggered, rather than through config) to get the same logging and error handling here?
https://github.com/Aam-Digital/aam-services/blob/76d30c009d557b41fcbded05d8d7843e76de9655/application/aam-backend-service/src/main/kotlin/com/aamdigital/aambackendservice/notification/core/create/app/AppCreateNotificationHandler.kt#L70

Reply:
I made the createDatabase function public, in case we need it dynamically.

Reply:
Is this a normal dependency to be injected then?

How do we replace this method with the CouchDbInitialized?
```
private fun ensureUserNotificationDatabaseExists(databaseName: String) {
        val databases = couchDbClient.allDatabases()
        if (!databases.contains(databaseName)) {
            couchDbClient.createDatabase(databaseName)
        }
    }
```

Let's do that right now, unless there is something that speaks against reusing the code

Reply:
Yes, its just a injection - I added the code to reuse the service for the dynamic db creation :)

Reply:
Thanks!